### PR TITLE
Validate firstname and lastname when user self signup

### PR DESCRIPTION
**Pull request by YasasRangika**
## Purpose
Validate first name and last name when user self signup.

## Goals
Fixes: https://github.com/wso2/product-apim/issues/11722

## Approach
Ignore if the first and last names having only white spaces.

### CODE REVIEW

File: apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp [hunk 1]
```diff
@@ -711,6 +711,17 @@
...
+                var firstname = $("#firstname").val();
+                var lastname = $("#lastname").val();
+
+                if (!firstname.trim() || !lastname.trim()) {
+                    error_msg.text("<%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle,
+                        "Invalid.firstname.or.lastname")%>");
+                    error_msg.show();
+                    $("html, body").animate({scrollTop: error_msg.offset().top}, 'slow');
+                    return false;
+                }
```
Comment (thivi) [inline apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp:723]: Can we have separate error messages for first name and last name?

Comment (jeradrutnam) [reply apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp:723]: And update the messages to something like: "First name cannot be empty, please enter a valid First name"

Comment (YasasRangika) [reply apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp]: Resolving with https://github.com/wso2/identity-apps/pull/2504/commits/0bc394fbffde8cdbd88de647a3e7cf0904c50bc7

File: apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties [hunk 2]
```diff
@@ -79,6 +79,8 @@ Cannot.process.email.confirmation.code.is.missing=Cannot process the email notif
 Cannot.process.submitted.confirmation.code=Invalid SMS OTP verification code submitted.
 Invalid.code=Provided confirmation code is invalid.
 Passwords.did.not.match.please.try.again=Passwords did not match. Please try again.
+Invalid.firstname=First name cannot be empty, please enter a valid first name.
```
Comment (thivi) [inline apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties:82]:
```suggestion
Invalid.firstname=First name cannot be empty. Please enter a valid first name.
```

File: apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties [hunk 3]
```diff
@@ -79,6 +79,8 @@ Cannot.process.email.confirmation.code.is.missing=Cannot process the email notif
...
+Invalid.lastname=Last name cannot be empty, please enter a valid last name.
```
Comment (thivi) [inline apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties:83]:
```suggestion
Invalid.lastname=Last name cannot be empty. Please enter a valid last name.
```

File: apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties (hunk 2 above)
Comment (YasasRangika) [reply apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties]: Added with https://github.com/wso2/identity-apps/pull/2504/commits/419b47ce153cebb41f468ecb92c3702182653184

File: apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties (hunk 3 above)
Comment (YasasRangika) [reply apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties]: Added with https://github.com/wso2/identity-apps/pull/2504/commits/419b47ce153cebb41f468ecb92c3702182653184

**Review by thivi — APPROVED** [review]

**Comment by YasasRangika** [comment]
need to revert or change according to https://github.com/wso2/product-apim/issues/12033

**Comment by lashinijay** [comment]
Product IS Issue - [https://github.com/wso2/product-is/issues/15528](https://github.com/wso2/product-is/issues/15528)

**Pull request merged**